test/http.js

```diff
@@ -360,64 +359,55 @@ describe('http', () => {
     test(
       'should serialize fetch-like response and call serializeHeaders',
       () => {
-        const headers = {
-          Authorization: ['Basic hoop-la', 'Advanced hoop-la']
-        }
-
-        const res = fetchMock.mock('http://swagger.io', {headers})
+        const response = new Response('data', {
+          status: 200,
+          headers: {
+            Authorization: 'Basic hoop-la, Advanced hoop-la',
+            'Content-Type': 'text/plain',
+          }
+        })
 
-        return fetch('http://swagger.io').then((_res) => { // eslint-disable-line no-undef
-          return serializeRes(_res, 'https://swagger.io')
-        }).then((resSerialize) => {
-          expect(resSerialize.headers).toEqual({authorization: ['Basic hoop-la', 'Advanced hoop-la']})
-        }).then(fetchMock.restore)
+        return serializeRes(response, 'https://swagger.io')
+          .then((serializedResponse) => {
+            expect(serializedResponse.headers).toEqual({
+              authorization: ['Basic hoop-la', 'Advanced hoop-la'],
+              'content-type': 'text/plain'
+            })
+          })
```

Comment:
we need `.catch` statements for all of these promise chains in test file. Node will fail missing catch in future

Reply:
What is say is true for `node.js`, but not true in our current context. We're node running our tests in pure `node.js`. We use testing framework [jest](https://jestjs.io/). And jest handles promises for us:

Excerpt from documentation:

> If your code uses promises, there is a more straightforward way to handle asynchronous tests. Return a promise from your test, and Jest will wait for that promise to resolve. If the promise is rejected, the test will automatically fail.

More [here](https://jestjs.io/docs/en/asynchronous#promises)

Reply:
ok for now. revisit later as needed.